add action to create PR-specific apps

mariogiampieri opened the pull request:
<!--- Provide a general summary of your changes in the Title above. -->

## Description
<!--- Describe your changes in detail -->
- Testing github actions to create pr-specific apps for testing before merging into main. Action steps include:
    - [x] forking current database and creating a new instance
    - [x] deploying a the `api` app
        - [x] passing the created database keys to api app
        - [x] being able to access db data via api (previously dealing with incorrect `db_url`)
    - [x] deploying the `app` app
        - [x] ensure that app has permissions under cors policy to access tiles / other bucket assets
        - [x] pass frontend app secrets 
    - [x] running any migrations       
    - [x] deleting these resources once the PR is merged in
    - [x] account for updates to pr re: redeploy

## Reviewers
<!--- Tag relevant reviewers. Expect the primary reviewer to "own" the review for this PR, -->
<!--- and the secondary to assist if the primary reviewer is delayed. -->

- Primary:
- Secondary:

## Checklist
<!--- Complete this checklist before asking for reviews. You can simply click the checkboxes once the markdown is rendered. -->
<!--- If not applicable, simply leave a reason why under the bullet point. -->
- [ ] Added/Updated related documentation (if applicable).
- [ ] Added/Updated related unit tests (if applicable).

## Screenshots (if applicable):

raphaellaude reviewed (changes requested):
Looks super good. I like how the action skips rebuilding the db + apps if they exist 🤗 . Just a few small comments

mariogiampieri commented on app/fly.toml:
```diff
@@ -11,9 +11,9 @@ primary_region = 'ewr'
+  auto_stop_machines = true
   auto_start_machines = true
-  min_machines_running = 1
+  min_machines_running = 2
```
yup yup, was just testing but agreed it's overkill for a pr branch

mariogiampieri commented on app/Dockerfile:
```diff
@@ -26,6 +26,17 @@ RUN npm ci --include=dev
+ENV NEXT_PUBLIC_S3_BUCKET_URL=${NEXT_PUBLIC_S3_BUCKET_URL}
+
+# echo env vars to check if they are set
+RUN echo $NEXT_PUBLIC_API_URL
+RUN echo $NEXT_PUBLIC_S3_BUCKET_URL
```
good call!

mariogiampieri commented on app/fly.toml:
```diff
@@ -11,9 +11,9 @@ primary_region = 'ewr'
   internal_port = 3000
   force_https = true
-  auto_stop_machines = 'stop'
+  auto_stop_machines = true
```
noted, and fine with having a different approach for prod vs testing later

mariogiampieri commented on .github/workflows/fly-deploy-pr.yml:
```diff
@@ -0,0 +1,173 @@
+                NEXT_PUBLIC_S3_BUCKET_URL=${{secrets.NEXT_PUBLIC_S3_BUCKET_URL}} 
+
+
+            - name: run database migrations
+              run: |
+                flyctl ssh console \
+                --app "pr-${{ github.event.number }}-${{ github.repository_owner }}-${{ github.event.repository.name }}-api" \
+                --command "alembic upgrade head"
+              working-directory: backend
```
i think that's right- you all did a great job with the backend documentation and people may end up having to futz with specific migrations if they aren't careful per your point. happy to leave for now, maybe worth changing the boilerplate PR text to state clearly what will happen on each PR?

raphaellaude reviewed (approved):
lgtm

mariogiampieri commented:
Rebased, added some documentation, and successfully tested shattering on the pr app. I manually removed the app so as to not waste resources but we should be good to go now.

raphaellaude reviewed (approved):
lgtm

raphaellaude commented:
@mariogiampieri I need this to test a feature real quick so I'm going to merge. Ty!
